lib/OpenCloud/ObjectStore/Resource/DataObject.php

```diff
@@ -354,6 +382,26 @@ public function delete($params = array())
     {
         return $this->getService()->getClient()->delete($this->getUrl())->send();
     }
+    
+    /**
+     * @param string $source Path (`container/object') of other object to symlink this object to
+     * @return \Guzzle\Http\Message\Response
+     */
+    public function symlink($source)
```

Comment:
Interesting thought, which I hadn't considered. Either direction the semantic is in however, there should be some some validation. 

Validation
========

In the case of the other file symlinking to this (createSymlinkFrom), then should there be checks

- the object does not exists or
- the object exists and can safely be made a symlink (i.e. getContentLength() == 0)

In the case of this file symlinking to the other file (createSymlinkTo), then there only needs to be the latter (as we implicitly know the object exists already)

- the object can safely be made a symlink (i.e. getContentLength() == 0)

Submission
=========

createSymlinkFrom has slighlty more complex logic:

- parse name of that other file (<container>/<resource>) into container and resource names
- get new container from the objectStoreService
- upload empty file with header

createSymlinkTo has slightly less complex logic:

- upload empty file with header
- set internal manifest

Other
=====

createSymlinkFrom follows Unix's `ln` convention

`ln -s $this $other`

Conclusion
=========

On balance my vote is for createSymlinkTo, but I've implemented both.